sky/data/storage.py

```diff
@@ -458,7 +492,8 @@ def __init__(self,
                  stores: Optional[Dict[StoreType, AbstractStore]] = None,
                  persistent: Optional[bool] = True,
                  mode: StorageMode = StorageMode.MOUNT,
-                 sync_on_reconstruction: bool = True) -> None:
+                 sync_on_reconstruction: bool = True,
+                 bucket_sub_path: Optional[str] = None) -> None:
```

Comment:
Can we rename this to `_bucket_sub_path`? Let's start with keeping this arg hidden and for internal use only. We can expose it publicly later if needed.

Reply:
Okay, I thought parameter names starting with an underscore would go against the coding guidelines.